homeassistant/components/ecobee/climate.py

```diff
@@ -741,6 +804,110 @@ def resume_program(self, resume_all):
         )
         self.update_without_throttle = True
 
+    def set_sensors_used_in_climate(
+        self, device_ids: list[str], preset_mode: str | None = None
+    ) -> None:
+        """Set the sensors used on a climate for a thermostat."""
+        if preset_mode is None:
+            preset_mode = self.preset_mode
+
+        # Check if climate is an available preset option.
+        elif preset_mode not in self._preset_modes.values():
+            if self.preset_modes:
+                raise ServiceValidationError(
+                    translation_domain=DOMAIN,
+                    translation_key="invalid_preset",
+                    translation_placeholders={
+                        "options": ", ".join(self._preset_modes.values())
+                    },
+                )
+
+        # Get device name from device id.
+        device_registry = dr.async_get(self.hass)
+        sensor_names: list[str] = []
+        sensor_ids: list[str] = []
+        for device_id in device_ids:
+            sensor_registry = device_registry.async_get(device_id)
```

Comment:
`DeviceRegistry.async_get` returns a device object, naming the return value `sensor_registry` is confusing IMO.
How about this:
```suggestion
            device = device_registry.async_get(device_id)
```